### ISSUE TITLE
Syncing over Wi-Fi hotspot

### ISSUE THREAD

**Issue by mehrdadn** [syncthing/syncthing-android#1308]
Hi, I wanted to request that the program allow syncing when the phone or the laptop is acting as a Wi-Fi hotspot. There really isn't any reason why a separate access point should be necessary for syncing, and it's frustrating to not be able to sync files when there isn't a Wi-Fi network available for this purpose. Thank you!

**Comment by capi** [comment]
Have you tried setting the "Start without network connection" setting (I think that's the English name) on the run conditions page (last entry)?

**Comment by mehrdadn** [comment]
Hi, yes, that only seems to work if I actually first disable my data connection, which I clearly don't want to do when I'm using the Wi-Fi hotspot.

**Comment by capi** [comment]
So, just to be clear, you have enabled "WiFi", "Mobile Data" and "No network connection" in the runtime settings and it does not start up? What's the reason that's shown in the disabled screen? Can you provide an adb log?

**Comment by mehrdadn** [comment]
Yup—although it should work even if I don't enable mobile data?

It actually seems to be rather flaky and sometimes work... eventually. Seems to happen if I run it enough times or wait long enough after changing the settings etc.

I even got into this bizarre situation where it was working but told me that it wasn't (see the right screenshot) despite having the settings you see on the left:

[![](https://i.imgur.com/rKeb4xJ.jpg)](https://i.imgur.com/rKeb4xJ.jpg)

ADB log might take me a bit longer—what kinds of entries should I look for/provide?

**Comment by capi** [comment]
If you are acting as hotspot, you are connected using mobile, not via the WiFi. So basically it should work with "Mobile" on.
I have noticed that the dialog is sometimes not properly dismissed, you should be able to get rid of it by entering "Change Settings" and returning (as a workaround).

**Comment by mehrdadn** [comment]
The thing is even if I don't have mobile data enabled and/or available, I should be able to use the hotspot to sync files. But it should also work if I do have data. The two are really completely unrelated to each other.

I've tried going to Change Settings and coming back and it doesn't always work. It worked maybe once I think. Why not just avoid popping it up in the first place?

**Comment by capi** [comment]
I think I finally understand, you are using the hotspot to use as an AdHoc-Wifi between notebook and the phone? I didn't even know that that worked.

**Comment by mehrdadn** [comment]
Yes! It's not obvious but you can actually totally do this, e.g. I use SSH or WinSCP to connect to the hotspot which is always 192.168.43.1. It lets me transfer files wirelessly without needing any other device.

**Comment by mehrdadn** [comment]
But at a higher level what I think is that you shouldn't try to support specific scenarios like this. The program should just bind to 0.0.0.0 to let the user use any network interface that might be available. I imagine that might be cellular data, Wi-Fi, mobile hotspot, some kind of Ethernet connection (I think those exist?), Bluetooth PAN (not sure but I'm guessing it's similar), or something more exotic.

**Comment by capi** [comment]
I think it would, the problem seems to be that this whole situation somehow confuses the run condition check. So as a workaround for you, you could disable "Always run in background" and start/stop Syncthing manually or via the intents via e.g. Tasker.

**Comment by mehrdadn** [comment]
Right but why is there a run condition check? Why not just let people run the application and simply let them know no network is detected?

**Comment by capi** [comment]
Most people want to run only under certain conditions (I do). If you check _all_ the options, it should always run. If not, then _that_ is a bug which we need to tackle.

**Comment by Catfriend1** [comment]
Hi,

To pick this up after some misunderstanding in another thread:
Which Android version are we talking about?
Phone is hotspot and laptop connects to it?
Does checking "run on metered wifi" help?
What is Syncthing's config.xml ? (Without keys and passwords) I'd check the bind addresses. 
If WiFi isn't detected at all what would seem like a bug in the run condition monitor, I'll make a test myself as this worked for me a few days ago. Thanks for your patience.

**Comment by mehrdadn** [comment]
I was on Oreo at the time, but this was back in April, so I don't know if anything has changed or not since I stopped using Syncthing. I'll try to run another test (not sure when I'll get around to it) and get back to you.

**Comment by mehrdadn** [comment]
@Catfriend1 I just tried it again and I think I got it working this time (via the direct IP address) when on Wi-Fi Hotspot (I enabled all modes including allowing Syncthing to connect under Airplane mode). I'm not sure what may have changed, if anything, but one thing I noticed is that, in my Pie, Android no longer uses 192.168.43.1 for the Wi-Fi Hotspot, but a random IP address under 192.168.43/24, which might be worth keeping in mind since it basically requires dynamic discovery.
I'll close this issue though since it seems to be working. Thanks!
